src/Sarif.Multitool/Rules/UrisMustBeValid.cs

```diff
@@ -29,69 +35,80 @@ protected override IEnumerable<string> FormatIds
+        {
+            if (result.WorkItemUris != null)
             {
-                Uri fileUri = new Uri(fileUriReference);
+                Uri[] workItemUris = result.WorkItemUris.ToArray();
+                string workItemUrisPointer = resultPointer.AtProperty(SarifPropertyName.WorkItemUris);
+
+                for (int i = 0; i < workItemUris.Length; ++i)
+                {
+                    AnalyzeUri(workItemUris[i], workItemUrisPointer.AtIndex(i));
+                }
             }
-            catch
+        }
+
+        protected override void Analyze(Rule rule, string rulePointer)
+        {
+            AnalyzeUri(rule.HelpUri, rulePointer.AtProperty(SarifPropertyName.HelpUri));
+        }
+
+        protected override void Analyze(Run run, string runPointer)
+        {
+            if (run.OriginalUriBaseIds != null)
             {
-                LogResult(
-                    filePointer,
-                    nameof(RuleResources.SARIF1003_Default),
-                    fileUriReference);
-            }
+                string originalUriBaseIdsPointer = runPointer.AtProperty(SarifPropertyName.OriginalUriBaseIds);
 
-            // Then check the "uri" property, if any, of the property value.
-            AnalyzeUri(fileData.Uri, filePointer);
+                foreach (string key in run.OriginalUriBaseIds.Keys)
+                {
+                    AnalyzeUri(run.OriginalUriBaseIds[key], originalUriBaseIdsPointer.AtProperty(key));
+                }
+            }
         }
 
-        protected override void Analyze(PhysicalLocation physicalLocation, string physicalLocationPointer)
+        protected override void Analyze(Tool tool, string toolPointer)
         {
-            AnalyzeUri(physicalLocation.Uri, physicalLocationPointer);
+            AnalyzeUri(tool.DownloadUri, toolPointer.AtProperty(SarifPropertyName.DownloadUri));
         }
 
-        protected override void Analyze(StackFrame frame, string framePointer)
+        protected override void Analyze(VersionControlDetails versionControlDetails, string versionControlDetailsPointer)
         {
-            AnalyzeUri(frame.Uri, framePointer);
+            AnalyzeUri(versionControlDetails.Uri, versionControlDetailsPointer.AtProperty(SarifPropertyName.Uri));
         }
 
-        protected override void Analyze(Rule rule, string rulePointer)
+        private void AnalyzeUri(Uri uri, string pointer)
         {
-            AnalyzeUri(rule.HelpUri, rulePointer, "helpUri");
+            AnalyzeUri(uri?.OriginalString, pointer);
         }
 
-        private void AnalyzeUri(
-            Uri uri,
-            string parentPointer,
-            string childPropertyName = SarifPropertyName.Uri)
+        private void AnalyzeUri(string uri, string pointer)
         {
             if (uri != null)
             {
                 try
                 {
-                    Uri fileUri = new Uri(uri.OriginalString);
+                    Uri fileUri = new Uri(uri);
```

Comment:
>Uri fileUri [](start = 20, length = 11)

Why aren't you using Uri.TryCreate here? Suggest doing so to prevent preventable exceptions.

Reply:
I didn't think of it. Much better. Will change.

---
In reply to: [210451903](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210451903) [](ancestors = 210451903)

Reply:
It doesn't work. `Uri.TryCreate` happily returns `true` and produces a `Uri` object whose `OriginalString` is `fi%le://whatever`.

---
In reply to: [210452152](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210452152) [](ancestors = 210452152,210451903)

Reply:
Broken API.

Try using Uri.IsWellFormedUriString FTW

---
In reply to: [210650717](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210650717) [](ancestors = 210650717,210452152,210451903)

Reply:
Works.

---
In reply to: [210653881](https://github.com/Microsoft/sarif-sdk/pull/1018#discussion_r210653881) [](ancestors = 210653881,210650717,210452152,210451903)